### PR TITLE
BUG: vtkPointSource random generator was seeded globally for streamli…

### PR DESCRIPTION
**Pull request by danlipsa**
…nes.

Seeds for streamlines were created only once globally. Now they
are created from scratch for each plot.

### CODE REVIEW

**Comment by danlipsa** [comment]
@doutriaux1 @aashish24 Please review. This fixes the streamline tests failures (in the new framework) seen by @doutriaux1

**Comment by danlipsa** [comment]
@doutriaux1 @aashish24 ping

**Comment by danlipsa** [comment]
@doutriaux1 @aashish24 This is what I get on my branch (and probably on master). Is that what you have?

Ran 229 tests, 18 failed (92.14% success)
Failed tests:
	tests/test_vcs_put_png_on_canvas.py
	tests/test_vcs_dump_json.py
	tests/test_vcs_draw_logo_on.py
	tests/test_vcs_gms_patterns_hatches.py
	tests/test_vcs_basic_text.py
	tests/test_vcs_iso_celine_part1.py
	tests/test_vcs_update_array.py
	tests/test_vcs_gms_animate_projected_plots.py
	tests/test_vcs_continents.py
	tests/test_vcs_drawLinesAndMarkersLegend.py
	tests/test_vcs_missing_opacity_gms.py
	tests/test_vcs_iso_celine_part2.py
	tests/test_vcs_box_custom_as_def_exts.py
	tests/test_vcs_line_point_extension.py
	tests/test_vcs_hatches_patterns.py
	tests/test_vcs_canvas_background.py
	tests/vtk_ui/test_vtk_ui_text_dimensions.py
	tests/vtk_ui/test_vtk_ui_widget_show_hide.py

**Comment by doutriaux1** [comment]
will run it tonight from home.

File: tests/test_vcs_vectors_robinson.py [hunk 1]
```diff
@@ -1,15 +1,41 @@
...
         p.type = "robinson"
         V.projection = p
-        self.x.plot(u,v,V, bg=self.bg)
-
-        fnm = "test_vcs_vectors_robinson.png"
+        self.x.plot(u,v,V, bg=1)
+        fnm += "_robinson.png"
         self.checkImage(fnm)
+
+    def test_vcs_vectors_robinson(self):
```
Comment (aashish24) [inline tests/test_vcs_vectors_robinson.py]: I think our naming convention is testVcsVectorsRobinson but I might be wrong here. We should document our code convention.

**Comment by aashish24** [comment]
The code mostly looks reasonable to me except the naming convention.

File: tests/test_vcs_vectors_robinson.py (hunk 1 above)
Comment (doutriaux1) [inline tests/test_vcs_vectors_robinson.py]: @danlipsa breaking in many tests right now will break the html output. Please revert for now and let's fix this at a later time.

**Comment by aashish24** [comment]
> @danlipsa breaking in many tests right now will break the html output. Please revert for now and let's fix this at a later time.

@doutriaux1 sorry I do not quite understand what you meant here. Are you saying that these changes are breaking other tests? Also not sure what needs to be reverted. Sorry perhaps @danlipsa might know better.

**Comment by doutriaux1** [comment]
@aashish24 no it's valid and won't break test, what will break is the html output with the nice diff picture/slider. Right now it's expecting one failure per script only. It is not going to be an easy fix to make it analyze output from multiple test from the same script and parse the mangled messages into each tests. So for now it's easier to keep it together and later on revisit this.

**Comment by doutriaux1** [comment]
@danlipsa i see the 1090 vs 1091 issue on my mac. Is that new? It wasn't here last week, did it get "fixed" in VTK itself or did you fix it in vcs?

**Comment by doutriaux1** [comment]
@danlipsa should we bite the bullet and from a Linux machine after all tests pass, change the default to 1090 and update the baselines?

**Comment by danlipsa** [comment]
@doutriaux1 Indeed there is the 1900 vs 1901 problem on macs:
https://gitlab.kitware.com/vtk/vtk/merge_requests/2567#note_237976
Sean McBride will take a look at this. Lets have all test pass on Linux and decide after that if its not fixed by then.

**Comment by danlipsa** [comment]
> So for now it's easier to keep it together and later on revisit this.
@doutriaux1 @aashish24 We can revisit this later. I changed this to have only one test.

File: tests/test_vcs_vectors_robinson.py (hunk 1 above)
Comment (danlipsa) [inline tests/test_vcs_vectors_robinson.py]: OK. We can revisit this after the release.

**Comment by doutriaux1** [comment]
ok testing branch on my linux (mac is broken because on 1090vs1091 issue)

**Comment by doutriaux1** [comment]
@danlipsa beside the failed vtk_ui test i now get this one (mac as well):
```
vtk-cdat-nox              7.1.0.2017.3.23.73fb094d17ccdd594a6abf9ff8732149c88dc2e6          py27_0    uvcdat
```
test error log:
```
Failed test: test_vcs_put_png_on_canvas on Thu Mar 30 11:34:12 2017
Log

E
======================================================================
ERROR: testPutPng (test_vcs_put_png_on_canvas.TestVCSPNG)
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/git/vcs/tests/test_vcs_put_png_on_canvas.py", line 19, in testPutPng
    self.putPNGOnCanvas()
  File "/git/vcs/tests/test_vcs_put_png_on_canvas.py", line 13, in putPNGOnCanvas
    self.x.put_png_on_canvas(self.png,zoom,xoffset,yoffset,units,fitToHeight,bg=self.bg)
  File "/git/vcs/vcs/Canvas.py", line 4923, in put_png_on_canvas
    **kargs)
  File "/git/vcs/vcs/VTKPlots.py", line 967, in put_png_on_canvas
    filename, zoom, xOffset, yOffset, units, fitToHeight, *args, **kargs)
  File "/git/vcs/vcs/VTKPlots.py", line 975, in put_img_on_canvas
    reader.SetFileName(filename)
AttributeError: 'NoneType' object has no attribute 'SetFileName'

----------------------------------------------------------------------
Ran 1 test in 0.042s

FAILED (errors=1)
```

**Comment by doutriaux1** [comment]
@danlipsa I updated your branch to download only the sample data that we need rather than everything every time! It does need cdat_info from nightly channel though

**Comment by danlipsa** [comment]
@doutriaux1 @aashish24 As it solves the streamline problem, should we merge this?

**Comment by doutriaux1** [comment]
Sure. I'll open an issue for the vtkpng.

**Pull request merged**